### PR TITLE
MWPW-147716: Disabled background scroll for filter dialog

### PR DESCRIPTION
**Pull request by mirafedas**
Disabled body scroll for merch-sidenav web component used in the catalog filter (on small screens it gets opened in a modal).

The AEM PSI check fails because the merch-card JS execution time is too long (around 1,057 ms), so the TBT score is too low.
The change in the PR is related only to merch-sidenav, which is not a culprit of low TBT, thus we can merge it, and should address the TBT score of the merch-card in a separate PR.

Corresponding tacocat PR: https://git.corp.adobe.com/wcms/tacocat.js/pull/613
Corresponding mas PR: https://github.com/adobecom/mas/pull/20

Resolves: [MWPW-147716](https://jira.corp.adobe.com/browse/MWPW-147716)

Test URLs:
Before: https://main--cc--adobecom.hlx.live/products/catalog?martech=off&georouting=off&mep=off
After: https://mwpw-147716-merch-sidenav-scroll-lock--cc--adobecom.hlx.live/products/animate/free-trial-download

Here is a page for testing for QA (remove the space, I added it to make the PSI check to ignore this link): 
https://mwpw-147716-merch-sidenav-scroll-lock --cc--adobecom.hlx.live/products/catalog?martech=off&georouting=off&mep=off

### CODE REVIEW

File: creativecloud/deps/merch-sidenav.js [hunk 1]
```diff
@@ -1,9 +1,9 @@
-// branch: catalog-regressions-4 commit: 90641e42ad012c2386133cbd2df855f468842b2d Wed, 29 May 2024 12:05:08 GMT
+// branch: main commit: dd5f72e1449ed667f48ef5be086043ddf9cc113d Mon, 24 Jun 2024 12:52:40 GMT
```
Comment (npeltier) [inline creativecloud/deps/merch-sidenav.js:1]: can you re-build from [ecf0570](https://github.com/adobecom/mas/commit/ecf0570abe50b3871d2e7b823cbdbdcc11f3e133) ?

File: creativecloud/deps/merch-sidenav.js [hunk 2]
```diff
@@ -381,6 +383,46 @@ customElements.define(
 var SPECTRUM_MOBILE_LANDSCAPE = "(max-width: 700px)";
```
[the remark this answers is not in the document]
Comment (mirafedas) [reply creativecloud/deps/merch-sidenav.js:383]: When we run `npm run build` in `mas`, all consts get transformed to vars :)

File: creativecloud/deps/merch-sidenav.js [hunk 3]
```diff
@@ -381,6 +383,46 @@ customElements.define(
...
 var TABLET_DOWN = "(max-width: 1199px)";
 
+// src/bodyScrollLock.js
+var isIosDevice = typeof window !== "undefined" && window.navigator && window.navigator.platform && (/iP(ad|hone|od)/.test(window.navigator.platform) || window.navigator.platform === "MacIntel" && window.navigator.maxTouchPoints > 1);
```
[the remark this answers is not in the document]
Comment (mirafedas) [reply creativecloud/deps/merch-sidenav.js:387]: Is it available in `mas`? Can't find it there

[the remark this answers is not in the document]
Comment (npeltier) [reply creativecloud/deps/merch-sidenav.js:387]: no, was just proposing to change the name to upper case as it's a const

[the remark this answers is not in the document]
Comment (mirafedas) [reply creativecloud/deps/merch-sidenav.js:387]: aha, got it, will open a new mas PR soon with this change

[the remark this answers is not in the document]
Comment (mirafedas) [reply creativecloud/deps/merch-sidenav.js:387]: MAS PR , ready for review: https://github.com/adobecom/mas/pull/20

**Review by yesil — APPROVED** [review]

**Comment by yesil** [comment]
@mirafedas please link your mas/tacocat ticket that fixes this, currently the sidenav dependency is built from your feature branch.

**Comment by mirafedas** [comment]
@yesil I will rebuild from the main branch when this mas PR is merged: https://github.com/adobecom/mas/pull/20

**Review by npeltier — APPROVED** [review]

**Review by 3ch023 — APPROVED** [review]

**Comment by aishwaryamathuria** [comment]
@npeltier @mirafedas  Could you help get this change verified and unblock it form merge ?

**Comment by mirafedas** [comment]
@aishwaryamathuria sure, asked @Roycethan to verify this

**Comment by Roycethan** [comment]
> https://mwpw-147716-merch-sidenav-scroll-lock--cc--adobecom.hlx.live/products/animate/free-trial-download

@aishwaryamathuria @mirafedas  Verified it (fixed -background scroll is now disabled). I did not got access to add labels to this repo. hence was not able to, but plz feel free to add it and also if possible grant access thanks

**Comment by aishwaryamathuria** [comment]
Adding verified label as confirmed by @Roycethan

**Review by aishwaryamathuria — APPROVED** [review]

**Pull request merged**